packages/react-core/src/components/AlertGroup/AlertGroupInline.tsx

```diff
@@ -18,8 +19,8 @@ export const AlertGroupInline: React.FunctionComponent<AlertGroupProps> = ({
     className={css(styles.alertGroup, className, isToast ? styles.modifiers.toast : '')}
     {...rest}
   >
-    {React.Children.toArray(children).map((Alert: React.ReactNode, index: number) => (
-      <li key={index}>{Alert}</li>
+    {React.Children.toArray(children).map((alert, index) => (
+      <li key={(alert as React.ReactElement<AlertProps>).props?.uniqueId || index}>{alert}</li>
```

Comment:
I've been thinking if it is the best solution to add the `uniqueId` prop as it is intended only for 'internal' use. Also, it may be a bit confusing for the user when the expected behavior of the alert depends on prop that is optional. 

What if we generated e.g. uuid and used it as key instead? This would however require either to use [a 3rd party package](https://www.npmjs.com/package/uuid) or write a generator, I've seen a couple of good implementations for example [here](https://stackoverflow.com/a/21963136). 

I'm aware that there are some drawbacks when using uuids. Therefore I'll be fine also with `uniqueId` prop, but wanted to bring up the other option too :)